Automattic/wp-calypso
# Site migration: Implement the site migration instruction i2

Closes https://github.com/Automattic/wp-calypso/issues/89760

## Proposed Changes
**Summary:**
* Add a hook to fully manage the pre-site migration (atomic transfer, plugin installation, migration key)
* Integratie the hook with the site migration instructions i2

This PR is the last part of #89760 where I am integrating the hooks created on the previous PR with the site migration instructions via the new hook.

## Testing Instructions
* Go to the `/start` 
* Enable the feature flag via `sessionStorage.setItem('flags', 'migration-flow/remove-processing-step') ` + reload
* Select a free plan
* Go through the migration steps until the checkout page
* After the payment check if you were redirected to the new instructions page `setup/site-migration/site-migration-instructions-i2` and you saw the new instructions with a spinner loader.
* After all loaders are complete, check if you can start the migration similar to the i1 flow

> [!NOTE]
> This PR is not handling errors during the critical errors during site migration process, they happen when:
> - Site transfer returns an unexpected error. 
> - We can't install or active the plugin after 10 retries

It will be implemented as part of another issue. 


### Video
https://github.com/Automattic/wp-calypso/assets/38718/580f77ec-005a-4ab4-90e0-8f5876f41945




## Pre-merge Checklist
